gridded_etl_tools/utils/store.py

```diff
@@ -167,7 +167,11 @@ def mapper(self, refresh: bool = False, **kwargs: dict) -> fsspec.mapping.FSMap:
             A `MutableMapping` which is the S3 key/value store
         """
         if refresh or not hasattr(self, "_mapper"):
-            self._mapper = s3fs.S3Map(root=self.path, s3=self.fs())
+            if self.dm.custom_output_path:
```

Comment:
This behavior at least needs to be documented in the docstring, since it will be confusing that `S3.path` returns a different path than the one used, and we should consider the overall design. I like that it uses an existing flag, and I understand that since it's an output path, it might make sense that the input path would be different, but that's not something we've done before on S3 and we don't have good documentation explaining it yet.

Reply:
Do you think this switch should be inserted within S3’s `path` property instead? 

I think it’s slightly more useful to be able to manipulate both the default path and the custom path but ultimately I’m not opposed to moving it over there.

Reply:
Long term, I think we would want separate input and output path values and maybe we would want them to be arguments to the extract and parse functions or maybe even we would want them to be arguments to separate store objects which were arguments to the extract and parse functions.

For now, I think putting this in S3.path makes sense because that's where it is in Local.path. Is this being used to implement a custom S3 path or a custom local path? If we need to support custom S3 paths, I would put the code in S3.path and I would change the docstring in DM.__init__ to state that the parameter overrides both the local and S3 paths. If this is meant to be for a custom local path, however, I would just create a new Local store object instead and get the path through there.

Reply:
Sounds good. I’ve moved this logic w/ S3 and Local paths and amended the docstring.